src/debugger/BreakpointManager.js

```diff
@@ -12,29 +12,32 @@
 import { PerFileBreakpointMap } from "./PerFileBreakpointMap.js";
 import { Breakpoint } from "./Breakpoint.js";
 import type { Breakpoint as BreakpointType } from "./types.js";
+import invariant from "./../invariant.js";
 
 // Storing BreakpointStores for all source files
 export class BreakpointManager {
   constructor() {
     this._breakpointMaps = new Map();
   }
-  _breakpointMaps: { [string]: PerFileBreakpointMap };
+  _breakpointMaps: Map<string, PerFileBreakpointMap>;
 
   addBreakpointMulti(breakpoints: Array<BreakpointType>) {
     this._doBreakpointsAction(breakpoints, this._addBreakpoint.bind(this));
   }
 
   _addBreakpoint(bp: BreakpointType) {
-    if (!(bp.filePath in this._breakpointMaps)) {
-      this._breakpointMaps[bp.filePath] = new PerFileBreakpointMap(bp.filePath);
+    if (!this._breakpointMaps.has(bp.filePath)) {
+      this._breakpointMaps.set(bp.filePath, new PerFileBreakpointMap(bp.filePath));
     }
-    let breakpointMap = this._breakpointMaps[bp.filePath];
+    let breakpointMap = this._breakpointMaps.get(bp.filePath);
+    invariant(breakpointMap);
     breakpointMap.addBreakpoint(bp.line, bp.column);
   }
 
   getBreakpoint(filePath: string, lineNum: number, columnNum: number = 0): void | Breakpoint {
-    if (filePath in this._breakpointMaps) {
-      let breakpointMap = this._breakpointMaps[filePath];
+    if (this._breakpointMaps.has(filePath)) {
```

Comment:
Instead of looking up filePath twice, just call get and check if the result is undefined.

Reply:
Changed.